.gitmodules

```diff
@@ -335,3 +335,7 @@
 [submodule "contrib/liburing"]
 	path = contrib/liburing
 	url = https://github.com/axboe/liburing
+[submodule "contrib/foundationdb"]
+	path = contrib/foundationdb
+	url = https://git.moqi.ai/mqdb/ecnu/foundationdb.git
```

Comment:
We don't allow submodules from outside of GitHub, btw the style check will tell about it.

Reply:
BTW this PR has ~8K lines for various FDB relate code, maybe it this implementation should do this on top of keeper/zookeeper first, and later if this will shows it's maturity and the throughput of the keeper/zookeeper will be a problem then FDB will added.

Or even better, add some interface for storing metadata elsewhere first (or maybe some special disk?), and then on top of it implement firstly keeper/zookeeper.

Reply:
UPD: 14k lines of code :)

Reply:
> ```shell
> $ find . -name '*FDB*' -or -path '*/FoundationDB/*' -or -path '*fdb*'  -type f | xargs cat | wc -l
> 14178
> ```
I think your search might include test files, which would make the code look like it has a lot more lines.
```
$ find ./src/ \( -name '*FDB*' -or -path '*/FoundationDB/*' -or -name '*fdb*' \) -type f | xargs cat | wc -l
8453
```

Reply:
> BTW this PR has ~8K lines for various FDB relate code, maybe it this implementation should do this on top of keeper/zookeeper first, and later if this will shows it's maturity and the throughput of the keeper/zookeeper will be a problem then FDB will added.
> 
> Or even better, add some interface for storing metadata elsewhere first (or maybe some special disk?), and then on top of it implement firstly keeper/zookeeper.

That's a really good suggestion because one of the reasons we initially adopted FDB was to address the bottleneck issue with Zookeeper. In addition, this PR only implements metadata separation in a single-node scenario (DatabaseAtomic), specifically for local disk read/write operations. Due to serialization concerns, the interface of Zookeeper isn't very suitable, so we separately implemented interfaces for reading and writing different metadata.

Reply:
>I think your search might include test files, which would make the code look like it has a lot more lines.

Yes, I thought about this, and decided that tests should be reviewed as well.

>so we separately implemented interfaces for reading and writing different metadata.

I'm talking about some common "interface", not about `if (fdb) loadFromFDB();`, something like there is for wide and compact parts, but for metadata. I.e. move common part into some interface class and has different implementations for local and zookeeper/keeper and/or fdb.

And maybe it can be implemented even as a IDisk interface, thought not sure that it will be enough (but maybe with transactions on top of the disk it could, but doubt that it will be clean).